controller/controller.go

```diff
@@ -62,6 +85,9 @@ type Controller interface {
 	// found, an error that you can identify using IsUnitNotFound is returned.
 	GetStatus(req Request) ([]fleet.UnitStatus, error)
 
+	// WaitForStatus waits for a group to reach the given status.
+	WaitForStatus(req Request, desired Status, closer <-chan struct{}) error
```

Comment:
We already talked about things like this. https://github.com/giantswarm/formica/pull/34#discussion-diff-54093471